src/core/metrics.cc

```diff
@@ -272,6 +272,36 @@ void impl::remove_registration(const metric_id& id) {
     }
 }
 
+void impl::replicate_metric_if_required(shared_ptr<registered_metric> metric) const {
+    auto full_name = metric->get_id().full_name();
+
+    auto matching_spec = std::find_if(
+        _metrics_to_replicate.begin(),
+        _metrics_to_replicate.end(),
+        [&full_name](const auto& spec) {
+          return full_name == spec.full_name;
+        });
+
+    if (matching_spec != _metrics_to_replicate.end()) {
+        const auto& metric_family = _value_map.at(full_name);
+        replicate_metric(*matching_spec, metric, metric_family);
+    }
+}
+
+void impl::remove_metric_replica_if_required(const metric_id& id) const {
+    auto matching_spec = std::find_if(
+        _metrics_to_replicate.begin(),
+        _metrics_to_replicate.end(),
+        [full_name = id.full_name()](const auto& spec) {
+          return full_name == spec.full_name;
+        });
+
+    if (matching_spec != _metrics_to_replicate.end()) {
```

Comment:
A little duplication here, which is fine, but if relying on `_metrics_to_replicate` being sorted, the `lower_bound` and equality check are probably just tricky enough to be worth extracting it out.

Reply:
I've changed to storing the specs in an unordered_set so we no longer do this lookup.